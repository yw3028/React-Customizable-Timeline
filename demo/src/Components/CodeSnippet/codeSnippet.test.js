import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';

import CodeSnippet from './CodeSnippet';

const renderComponent = (colors, font, shapes, animation) =>
  render(<CodeSnippet colors={colors} font={font} shapes={shapes} animation={animation} />);

describe('AnimationToggle component', () => {
  afterEach(() => {
    cleanup();
  });

  test('render correctly', () => {
    const component = renderComponent(defaultColors, 'chivo', defaultShapes, true);
    expect(component.baseElement).toMatchSnapshot();
  });

  it("Doesn't render children elements", () => {
    render(
      <CodeSnippet colors={defaultColors} font={'chivo'} shapes={defaultShapes} animation={false}>
        <p>LisaPisa</p>
      </CodeSnippet>,
    );

    expect(screen.queryByText(/LisaPisa/i)).toBeFalsy();
  });
});

const defaultColors = {
  primaryColor: '#625261',
  secondaryColor: '#F5F5DC',
  backgroundColor: '#FFFFFF',
};

const defaultShapes = {
  titleShape: 'diamond',
  dotShape: 'circle',
  lineShape: 'dotted',
};
