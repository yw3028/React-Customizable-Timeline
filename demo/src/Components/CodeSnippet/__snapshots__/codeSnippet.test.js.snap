// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimationToggle component render correctly 1`] = `
<body>
  <div>
    <pre
      class=" language-javascript"
    >
      <code
        class=" language-javascript"
      >
        <span
          class="token operator"
        >
          &lt;
        </span>
        Timeline
  data
        <span
          class="token operator"
        >
          =
        </span>
        <span
          class="token punctuation"
        >
          {
        </span>
        data
        <span
          class="token punctuation"
        >
          }
        </span>
        
  primaryDarkColor
        <span
          class="token operator"
        >
          =
        </span>
        <span
          class="token string"
        >
          '#625261'
        </span>
         
  primaryLightColor
        <span
          class="token operator"
        >
          =
        </span>
        <span
          class="token string"
        >
          '#F5F5DC'
        </span>
         
  titleShape
        <span
          class="token operator"
        >
          =
        </span>
        <span
          class="token string"
        >
          'diamond'
        </span>
         
  dotShape
        <span
          class="token operator"
        >
          =
        </span>
        <span
          class="token string"
        >
          'circle'
        </span>
         
  lineStyle
        <span
          class="token operator"
        >
          =
        </span>
        <span
          class="token string"
        >
          'dotted'
        </span>
         
  primaryFont
        <span
          class="token operator"
        >
          =
        </span>
        <span
          class="token string"
        >
          'chivo'
        </span>
         
  animation
        <span
          class="token operator"
        >
          =
        </span>
        <span
          class="token punctuation"
        >
          {
        </span>
        <span
          class="token boolean"
        >
          true
        </span>
        <span
          class="token punctuation"
        >
          }
        </span>
        

        <span
          class="token operator"
        >
          /
        </span>
        <span
          class="token operator"
        >
          &gt;
        </span>
      </code>
    </pre>
  </div>
</body>
`;
